### ISSUE TITLE
Numeric facet values should not be treated as text strings

### ISSUE THREAD

**Issue by Tasilee** [AtlasOfLivingAustralia/spatial-hub#185]
**Describe the bug**
Currently, trait values are all being treated as text strings. This results in many more classes than is easy to digest when viewed in the Spatial Portal. We need to identify the traits with numeric values and break them into a maximum of 7 ranked classes.

**To Reproduce**
1. Go to 'Add to Map | Species | Use species List | Acacia characters'
1. Click on 'Facet in the legend' and select "MedianLefLength"
1. 24 classes are displayed

**Expected behavior**
Identify numeric traits and class them into ~7 ranked categories, probably equal interval.

**Comment by Tasilee** [comment]
This is a specific example of what now appears to be a more general issue in the biocahe.  A quick look at Darwin Core terms suggests the following need to be treated as numeric (ignore anything non-numeric)

individualCount
organismQuantity
startDayOfYear
endDayOfyear
year
month
day
sampleSizeValue
minimumElevationInMeters
maximumElevationInMeters
minimumDepthInMeters
maximumDepthInMeters
minimumDistanceAboveSurfaceInMeters
maximumDistanceAboveSurfaceInMeters
decimalLatitude
decimalLongitude
coordinateUncertaintyInMeters
coordinatePrecision
pointRadiusSpatialFit
footprintSpatialFit
namePublishedInYear

**Comment by charvolant** [comment]
individualCount is not always a straight numeric value. It can be something like "10-20" or "5 male, 3 female" or some rambling sentence that describes the observer's state of mind at the time.

organismQuantity can also be a letter eg. r on the BraunBlanquetScale from the DwC examples.

I'm not sure how these should be handled.

**Comment by Tasilee** [comment]
Thanks Doug. While text may be found in dwc:individualCount, this is not what Darwin Core suggests

"The number of individuals represented present at the time of the Occurrence. E.g., 	0, 1, 25.

I would certainly anticipate non-numeric characters where they should not be, so as I said above, such text should be ignored, unless we want to get into some serious parsing? Ditto for all noted numeric terms.

This is very reminiscent of https://github.com/tdwg/dwc-qa/tree/master/data. In short, from a user-perspective, we are better to render and use numerics as such rather than interpret numeric values as text strings.

**Comment by temi** [comment]
https://upsource.ala.org.au/spatial-hub/review/SH-CR-11

**Comment by Tasilee** [comment]
1. If I facet on dwc:year, the chart looks good except I am unsure what the top class ("* TO *") represents. In my example, this had 15 records. Are they record without an interpretable year value?
1. Still can't use slider bar effectively in "View in larger window".
1. Above, I noted only those Darwin Core terms that are numeric, but there are other terms that we also need to handle in the same way. E.g., ala:outlierLayerCount which is handled as 
* TO * | 494 |  
1 TO 3 | 4
In other words, you can't filter out records that have an outlier layer count of say "3". We need a field in our new Vocabulary that states this. This information is I think stored in https://bie.ala.org.au/ws/admin/indexFields as "dataType: int" etc. but this seems a subset, maybe i180?
1. "Depth" response as "depth.novalue" is probably ok, maybe a tad cryptic for the average user.
1.  Defaulting to equal ranges in histogram classes is good.
1. Rollovers on charts is good.

**Comment by temi** [comment]
1. It is for those with not year value. Yes, * TO * does not make much sense. It now should say "EXCLUDE * TO *".
2. This is fixed in the latest change.
3.  There is a bug in this specific case. But excluding specific numbers in a range looks to me like a new issue.
4. Need to change how messages are shown.
